packages/builders/src/messages/embed/UnsafeEmbed.ts

```diff
@@ -126,7 +179,8 @@ export class UnsafeEmbed implements APIEmbed, JSONEncodable<APIEmbed> {
 	 * @param fields The fields to add
 	 */
 	public addFields(...fields: APIEmbedField[]): this {
-		this.fields.push(...UnsafeEmbed.normalizeFields(...fields));
+		this.data.fields ??= [];
+		this.data.fields.push(...UnsafeEmbed.normalizeFields(...fields));
```

Comment:
We can do this to make this faster:

```suggestion
		fields = UnsafeEmbed.normalizeFields(...fields);
	
		if (this.data.fields) this.data.fields.push(...fields);
		else this.data.fields = fields;
```